### ISSUE TITLE
Entry info endpoints have no `id`, `type`, or `attributes`

### ISSUE THREAD

**Issue by ml-evs** [Materials-Consortia/OPTIMADE#470]
I thought I'd also opened this issue a while ago, but maybe not...

As part of the OPTIMADE file format discussions, we discovered that it is currently quite hard to use the `info/structures` (for e.g.,) endpoint response in a place where is it not attached to its URL. From the response alone it is impossible to tell which entry type the info corresponds to. We kind of break the JSON:API format by allowing entry info endpoints to present just a flat dictionary under `data`, with no `attributes`, `id` or `type`, so I think I would consider this a specification bug.

By simply adding the `id` and `type` fields to be `structures` and `info` respectively, we can then formulate a line in a file that is fully descriptive, i.e., that these properties are describing a structures entry type. As this also affects all the properties definition work (cc @rartino), I don't suggest that we force all other fields to be nested under `attributes`, though perhaps for OPTIMADE 2.0 this would be nice. @giovannipizzi also suggested perhaps allowing that at this stage to nest these fields under `attributes` and deprecating the current way in 1.2.

**Comment by rartino** [comment]
Indeed, these endpoints are broken and not JSON API compliant at all. But, to allow implementations to provide everything under attributes instead of data would clearly break clients. Perhaps we shouldn't be overly afraid to move to v2.0 now that so much other things are added as well (partial data and property definitions).

**Comment by ml-evs** [comment]
> Perhaps we shouldn't be overly afraid to move to v2.0 now that so much other things are added as well (partial data and property definitions).

Yes, I was thinking this in the workshop, or at least that 2.0 could follow fairly soon after 1.2 (then we write another paper? :sweat_smile:). I think at the very least I would like to write a PR that *allows* `id` and `type` to be provided alongside the non-standard top-level fields in 1.2, that way we can write the file serialisation with 1.2 for servers that support it, and "impute" it for servers that don't (i.e., my client can use the fact it is from an `info/structures` to populate the field. I will try to make a PR now...

**Comment by ml-evs** [comment]
Closed by #472, though we will need to remember to enact our proposed deprecation at some point (probably 2.0...)
